Feat: AWS Resilience Hub App resource

briancain opened the pull request:
Hello AWS TF Provider maintainers! I have another pull request for you. This PR fills a gap in the existing Resilience Hub service for the App resource, which has not yet been covered by the official AWS provider.

<!---
See what makes a good Pull Request at: https://hashicorp.github.io/terraform-provider-aws/raising-a-pull-request/
--->

<!-- heimdall_github_prtemplate:grc-pci_dss-2024-01-05 -->

## Rollback Plan

If a change needs to be reverted, we will publish an updated version of the library.

## Changes to Security Controls

Are there any changes to security controls (access controls, encryption, logging) in this pull request? If so, explain.

N/A

### Description
<!---
Please provide a helpful description of what change this pull request will introduce.
--->

This commit adds a gap in the AWS Resilience Hub service: an app resource. It includes the schema required to define and manage that app. It is a simple resource with only a few fields. This commit does add more unit and acceptance tests to verify the change.

### Relations

Closes https://github.com/hashicorp/terraform-provider-aws/issues/35803.
Closes https://github.com/hashicorp/terraform-provider-aws/pull/25123.

### References
<!---
Optionally, provide any helpful references that may help the reviewer(s).
--->

* https://aws.amazon.com/resilience-hub/
* https://docs.aws.amazon.com/resilience-hub/latest/userguide/what-is.html
* https://github.com/aws/aws-sdk-go-v2/tree/main/service/resiliencehub
* [Existing resource: "aws_resiliencehub_resiliency_policy"](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/resiliencehub_resiliency_policy)

### Output from Acceptance Testing
<!--
Replace TestAccXXX with a pattern that matches the tests affected by this PR.

Replace ec2 with the service package corresponding to your tests.

For more information on the `-run` flag, see the `go test` documentation at https://tip.golang.org/cmd/go/#hdr-Testing_flags.
-->

```console
$ AWS_PROFILE=default TF_ACC=1 go test ./internal/service/resiliencehub/... -v -short -parallel 1 -timeout 30m
2025/08/28 13:43:09 Creating Terraform AWS Provider (SDKv2-style)...
2025/08/28 13:43:09 Initializing Terraform AWS Provider (SDKv2-style)...
=== RUN   TestAccResilienceHubApp_tags
=== PAUSE TestAccResilienceHubApp_tags
=== RUN   TestAccResilienceHubApp_tags_null
=== PAUSE TestAccResilienceHubApp_tags_null
=== RUN   TestAccResilienceHubApp_tags_EmptyMap
=== PAUSE TestAccResilienceHubApp_tags_EmptyMap
=== RUN   TestAccResilienceHubApp_tags_AddOnUpdate
=== PAUSE TestAccResilienceHubApp_tags_AddOnUpdate
=== RUN   TestAccResilienceHubApp_tags_EmptyTag_OnCreate
=== PAUSE TestAccResilienceHubApp_tags_EmptyTag_OnCreate
=== RUN   TestAccResilienceHubApp_tags_EmptyTag_OnUpdate_Add
=== PAUSE TestAccResilienceHubApp_tags_EmptyTag_OnUpdate_Add
=== RUN   TestAccResilienceHubApp_tags_EmptyTag_OnUpdate_Replace
=== PAUSE TestAccResilienceHubApp_tags_EmptyTag_OnUpdate_Replace
=== RUN   TestAccResilienceHubApp_tags_DefaultTags_providerOnly
=== PAUSE TestAccResilienceHubApp_tags_DefaultTags_providerOnly
=== RUN   TestAccResilienceHubApp_tags_DefaultTags_nonOverlapping
=== PAUSE TestAccResilienceHubApp_tags_DefaultTags_nonOverlapping
=== RUN   TestAccResilienceHubApp_tags_DefaultTags_overlapping
=== PAUSE TestAccResilienceHubApp_tags_DefaultTags_overlapping
=== RUN   TestAccResilienceHubApp_tags_DefaultTags_updateToProviderOnly
=== PAUSE TestAccResilienceHubApp_tags_DefaultTags_updateToProviderOnly
=== RUN   TestAccResilienceHubApp_tags_DefaultTags_updateToResourceOnly
=== PAUSE TestAccResilienceHubApp_tags_DefaultTags_updateToResourceOnly
=== RUN   TestAccResilienceHubApp_tags_DefaultTags_emptyResourceTag
=== PAUSE TestAccResilienceHubApp_tags_DefaultTags_emptyResourceTag
=== RUN   TestAccResilienceHubApp_tags_DefaultTags_emptyProviderOnlyTag
=== PAUSE TestAccResilienceHubApp_tags_DefaultTags_emptyProviderOnlyTag
=== RUN   TestAccResilienceHubApp_tags_DefaultTags_nullOverlappingResourceTag
=== PAUSE TestAccResilienceHubApp_tags_DefaultTags_nullOverlappingResourceTag
=== RUN   TestAccResilienceHubApp_tags_DefaultTags_nullNonOverlappingResourceTag
=== PAUSE TestAccResilienceHubApp_tags_DefaultTags_nullNonOverlappingResourceTag
=== RUN   TestAccResilienceHubApp_tags_ComputedTag_OnCreate
=== PAUSE TestAccResilienceHubApp_tags_ComputedTag_OnCreate
=== RUN   TestAccResilienceHubApp_tags_ComputedTag_OnUpdate_Add
=== PAUSE TestAccResilienceHubApp_tags_ComputedTag_OnUpdate_Add
=== RUN   TestAccResilienceHubApp_tags_ComputedTag_OnUpdate_Replace
=== PAUSE TestAccResilienceHubApp_tags_ComputedTag_OnUpdate_Replace
=== RUN   TestAccResilienceHubApp_tags_IgnoreTags_Overlap_DefaultTag
=== PAUSE TestAccResilienceHubApp_tags_IgnoreTags_Overlap_DefaultTag
=== RUN   TestAccResilienceHubApp_tags_IgnoreTags_Overlap_ResourceTag
=== PAUSE TestAccResilienceHubApp_tags_IgnoreTags_Overlap_ResourceTag
=== RUN   TestAccResilienceHubApp_basic
=== PAUSE TestAccResilienceHubApp_basic
=== RUN   TestAccResilienceHubApp_terraformSource
=== PAUSE TestAccResilienceHubApp_terraformSource
=== RUN   TestAccResilienceHubApp_complete
=== PAUSE TestAccResilienceHubApp_complete
=== RUN   TestAccResilienceHubApp_update
=== PAUSE TestAccResilienceHubApp_update
=== RUN   TestAccResilienceHubResiliencyPolicy_tags
=== PAUSE TestAccResilienceHubResiliencyPolicy_tags
=== RUN   TestAccResilienceHubResiliencyPolicy_tags_null
=== PAUSE TestAccResilienceHubResiliencyPolicy_tags_null
=== RUN   TestAccResilienceHubResiliencyPolicy_tags_EmptyMap
=== PAUSE TestAccResilienceHubResiliencyPolicy_tags_EmptyMap
=== RUN   TestAccResilienceHubResiliencyPolicy_tags_AddOnUpdate
=== PAUSE TestAccResilienceHubResiliencyPolicy_tags_AddOnUpdate
=== RUN   TestAccResilienceHubResiliencyPolicy_tags_EmptyTag_OnCreate
=== PAUSE TestAccResilienceHubResiliencyPolicy_tags_EmptyTag_OnCreate
=== RUN   TestAccResilienceHubResiliencyPolicy_tags_EmptyTag_OnUpdate_Add
=== PAUSE TestAccResilienceHubResiliencyPolicy_tags_EmptyTag_OnUpdate_Add
=== RUN   TestAccResilienceHubResiliencyPolicy_tags_EmptyTag_OnUpdate_Replace
=== PAUSE TestAccResilienceHubResiliencyPolicy_tags_EmptyTag_OnUpdate_Replace
=== RUN   TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_providerOnly
=== PAUSE TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_providerOnly
=== RUN   TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_nonOverlapping
=== PAUSE TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_nonOverlapping
=== RUN   TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_overlapping
=== PAUSE TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_overlapping
=== RUN   TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_updateToProviderOnly
=== PAUSE TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_updateToProviderOnly
=== RUN   TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_updateToResourceOnly
=== PAUSE TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_updateToResourceOnly
=== RUN   TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_emptyResourceTag
=== PAUSE TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_emptyResourceTag
=== RUN   TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_emptyProviderOnlyTag
=== PAUSE TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_emptyProviderOnlyTag
=== RUN   TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_nullOverlappingResourceTag
=== PAUSE TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_nullOverlappingResourceTag
=== RUN   TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_nullNonOverlappingResourceTag
=== PAUSE TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_nullNonOverlappingResourceTag
=== RUN   TestAccResilienceHubResiliencyPolicy_tags_ComputedTag_OnCreate
=== PAUSE TestAccResilienceHubResiliencyPolicy_tags_ComputedTag_OnCreate
=== RUN   TestAccResilienceHubResiliencyPolicy_tags_ComputedTag_OnUpdate_Add
=== PAUSE TestAccResilienceHubResiliencyPolicy_tags_ComputedTag_OnUpdate_Add
=== RUN   TestAccResilienceHubResiliencyPolicy_tags_ComputedTag_OnUpdate_Replace
=== PAUSE TestAccResilienceHubResiliencyPolicy_tags_ComputedTag_OnUpdate_Replace
=== RUN   TestAccResilienceHubResiliencyPolicy_tags_IgnoreTags_Overlap_DefaultTag
=== PAUSE TestAccResilienceHubResiliencyPolicy_tags_IgnoreTags_Overlap_DefaultTag
=== RUN   TestAccResilienceHubResiliencyPolicy_tags_IgnoreTags_Overlap_ResourceTag
=== PAUSE TestAccResilienceHubResiliencyPolicy_tags_IgnoreTags_Overlap_ResourceTag
=== RUN   TestAccResilienceHubResiliencyPolicy_basic
=== PAUSE TestAccResilienceHubResiliencyPolicy_basic
=== RUN   TestAccResilienceHubResiliencyPolicy_dataLocationConstraint
=== PAUSE TestAccResilienceHubResiliencyPolicy_dataLocationConstraint
=== RUN   TestAccResilienceHubResiliencyPolicy_description
=== PAUSE TestAccResilienceHubResiliencyPolicy_description
=== RUN   TestAccResilienceHubResiliencyPolicy_name
=== PAUSE TestAccResilienceHubResiliencyPolicy_name
=== RUN   TestAccResilienceHubResiliencyPolicy_policy
=== PAUSE TestAccResilienceHubResiliencyPolicy_policy
=== RUN   TestAccResilienceHubResiliencyPolicy_policyWithRegion
=== PAUSE TestAccResilienceHubResiliencyPolicy_policyWithRegion
=== RUN   TestAccResilienceHubResiliencyPolicy_tier
=== PAUSE TestAccResilienceHubResiliencyPolicy_tier
=== RUN   TestAccResilienceHubResiliencyPolicy_disappears
=== PAUSE TestAccResilienceHubResiliencyPolicy_disappears
=== RUN   TestAccResilienceHubResiliencyPolicy_upgradeV6_0_0
=== PAUSE TestAccResilienceHubResiliencyPolicy_upgradeV6_0_0
=== RUN   TestEndpointConfiguration
=== RUN   TestEndpointConfiguration/package_name_endpoint_config_overrides_aws_service_envvar
=== RUN   TestEndpointConfiguration/package_name_endpoint_config_overrides_base_envvar
=== RUN   TestEndpointConfiguration/service_aws_envvar_overrides_service_config_file
=== RUN   TestEndpointConfiguration/use_fips_config_with_package_name_endpoint_config
=== RUN   TestEndpointConfiguration/package_name_endpoint_config_overrides_base_config_file
=== RUN   TestEndpointConfiguration/service_config_file
=== RUN   TestEndpointConfiguration/no_config
=== RUN   TestEndpointConfiguration/package_name_endpoint_config_overrides_service_config_file
=== RUN   TestEndpointConfiguration/service_aws_envvar_overrides_base_envvar
=== RUN   TestEndpointConfiguration/base_endpoint_envvar
=== RUN   TestEndpointConfiguration/base_endpoint_envvar_overrides_service_config_file
=== RUN   TestEndpointConfiguration/package_name_endpoint_config
=== RUN   TestEndpointConfiguration/service_aws_envvar
=== RUN   TestEndpointConfiguration/service_aws_envvar_overrides_base_config_file
=== RUN   TestEndpointConfiguration/base_endpoint_envvar_overrides_base_config_file
=== RUN   TestEndpointConfiguration/service_config_file_overrides_base_config_file
=== RUN   TestEndpointConfiguration/base_endpoint_config_file
=== RUN   TestEndpointConfiguration/use_fips_config
--- PASS: TestEndpointConfiguration (0.35s)
    --- PASS: TestEndpointConfiguration/package_name_endpoint_config_overrides_aws_service_envvar (0.03s)
    --- PASS: TestEndpointConfiguration/package_name_endpoint_config_overrides_base_envvar (0.03s)
    --- PASS: TestEndpointConfiguration/service_aws_envvar_overrides_service_config_file (0.01s)
    --- PASS: TestEndpointConfiguration/use_fips_config_with_package_name_endpoint_config (0.03s)
    --- PASS: TestEndpointConfiguration/package_name_endpoint_config_overrides_base_config_file (0.03s)
    --- PASS: TestEndpointConfiguration/service_config_file (0.01s)
    --- PASS: TestEndpointConfiguration/no_config (0.01s)
    --- PASS: TestEndpointConfiguration/package_name_endpoint_config_overrides_service_config_file (0.03s)
    --- PASS: TestEndpointConfiguration/service_aws_envvar_overrides_base_envvar (0.01s)
    --- PASS: TestEndpointConfiguration/base_endpoint_envvar (0.01s)
    --- PASS: TestEndpointConfiguration/base_endpoint_envvar_overrides_service_config_file (0.02s)
    --- PASS: TestEndpointConfiguration/package_name_endpoint_config (0.02s)
    --- PASS: TestEndpointConfiguration/service_aws_envvar (0.01s)
    --- PASS: TestEndpointConfiguration/service_aws_envvar_overrides_base_config_file (0.01s)
    --- PASS: TestEndpointConfiguration/base_endpoint_envvar_overrides_base_config_file (0.01s)
    --- PASS: TestEndpointConfiguration/service_config_file_overrides_base_config_file (0.01s)
    --- PASS: TestEndpointConfiguration/base_endpoint_config_file (0.01s)
    --- PASS: TestEndpointConfiguration/use_fips_config (0.01s)
=== CONT  TestAccResilienceHubResiliencyPolicy_disappears
--- PASS: TestAccResilienceHubResiliencyPolicy_disappears (14.66s)
=== CONT  TestAccResilienceHubApp_tags
--- PASS: TestAccResilienceHubApp_tags (62.36s)
=== CONT  TestAccResilienceHubResiliencyPolicy_tier
--- PASS: TestAccResilienceHubResiliencyPolicy_tier (30.54s)
=== CONT  TestAccResilienceHubResiliencyPolicy_policyWithRegion
--- PASS: TestAccResilienceHubResiliencyPolicy_policyWithRegion (44.19s)
=== CONT  TestAccResilienceHubResiliencyPolicy_policy
--- PASS: TestAccResilienceHubResiliencyPolicy_policy (45.10s)
=== CONT  TestAccResilienceHubResiliencyPolicy_name
--- PASS: TestAccResilienceHubResiliencyPolicy_name (30.92s)
=== CONT  TestAccResilienceHubResiliencyPolicy_description
--- PASS: TestAccResilienceHubResiliencyPolicy_description (31.14s)
=== CONT  TestAccResilienceHubResiliencyPolicy_dataLocationConstraint
--- PASS: TestAccResilienceHubResiliencyPolicy_dataLocationConstraint (30.56s)
=== CONT  TestAccResilienceHubResiliencyPolicy_basic
--- PASS: TestAccResilienceHubResiliencyPolicy_basic (16.03s)
=== CONT  TestAccResilienceHubResiliencyPolicy_tags_IgnoreTags_Overlap_ResourceTag
--- PASS: TestAccResilienceHubResiliencyPolicy_tags_IgnoreTags_Overlap_ResourceTag (39.56s)
=== CONT  TestAccResilienceHubResiliencyPolicy_tags_IgnoreTags_Overlap_DefaultTag
--- PASS: TestAccResilienceHubResiliencyPolicy_tags_IgnoreTags_Overlap_DefaultTag (35.17s)
=== CONT  TestAccResilienceHubApp_basic
--- PASS: TestAccResilienceHubApp_basic (17.17s)
=== CONT  TestAccResilienceHubApp_tags_IgnoreTags_Overlap_ResourceTag
--- PASS: TestAccResilienceHubApp_tags_IgnoreTags_Overlap_ResourceTag (41.66s)
=== CONT  TestAccResilienceHubApp_tags_IgnoreTags_Overlap_DefaultTag
--- PASS: TestAccResilienceHubApp_tags_IgnoreTags_Overlap_DefaultTag (39.16s)
=== CONT  TestAccResilienceHubApp_tags_ComputedTag_OnUpdate_Replace
--- PASS: TestAccResilienceHubApp_tags_ComputedTag_OnUpdate_Replace (34.31s)
=== CONT  TestAccResilienceHubApp_tags_ComputedTag_OnUpdate_Add
--- PASS: TestAccResilienceHubApp_tags_ComputedTag_OnUpdate_Add (33.93s)
=== CONT  TestAccResilienceHubApp_tags_ComputedTag_OnCreate
--- PASS: TestAccResilienceHubApp_tags_ComputedTag_OnCreate (21.34s)
=== CONT  TestAccResilienceHubApp_terraformSource
--- PASS: TestAccResilienceHubApp_terraformSource (18.93s)
=== CONT  TestAccResilienceHubApp_tags_DefaultTags_nullNonOverlappingResourceTag
--- PASS: TestAccResilienceHubApp_tags_DefaultTags_nullNonOverlappingResourceTag (19.93s)
=== CONT  TestAccResilienceHubApp_tags_DefaultTags_nullOverlappingResourceTag
--- PASS: TestAccResilienceHubApp_tags_DefaultTags_nullOverlappingResourceTag (19.65s)
=== CONT  TestAccResilienceHubResiliencyPolicy_tags_ComputedTag_OnUpdate_Replace
--- PASS: TestAccResilienceHubResiliencyPolicy_tags_ComputedTag_OnUpdate_Replace (31.06s)
=== CONT  TestAccResilienceHubApp_tags_DefaultTags_emptyProviderOnlyTag
--- PASS: TestAccResilienceHubApp_tags_DefaultTags_emptyProviderOnlyTag (18.86s)
=== CONT  TestAccResilienceHubResiliencyPolicy_tags_ComputedTag_OnUpdate_Add
--- PASS: TestAccResilienceHubResiliencyPolicy_tags_ComputedTag_OnUpdate_Add (30.61s)
=== CONT  TestAccResilienceHubApp_tags_DefaultTags_emptyResourceTag
--- PASS: TestAccResilienceHubApp_tags_DefaultTags_emptyResourceTag (19.69s)
=== CONT  TestAccResilienceHubResiliencyPolicy_tags_ComputedTag_OnCreate
--- PASS: TestAccResilienceHubResiliencyPolicy_tags_ComputedTag_OnCreate (20.07s)
=== CONT  TestAccResilienceHubApp_tags_DefaultTags_updateToResourceOnly
--- PASS: TestAccResilienceHubApp_tags_DefaultTags_updateToResourceOnly (29.72s)
=== CONT  TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_nullNonOverlappingResourceTag
--- PASS: TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_nullNonOverlappingResourceTag (17.34s)
=== CONT  TestAccResilienceHubApp_tags_DefaultTags_updateToProviderOnly
--- PASS: TestAccResilienceHubApp_tags_DefaultTags_updateToProviderOnly (29.97s)
=== CONT  TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_nullOverlappingResourceTag
--- PASS: TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_nullOverlappingResourceTag (17.89s)
=== CONT  TestAccResilienceHubApp_tags_DefaultTags_overlapping
--- PASS: TestAccResilienceHubApp_tags_DefaultTags_overlapping (49.13s)
=== CONT  TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_emptyProviderOnlyTag
--- PASS: TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_emptyProviderOnlyTag (17.25s)
=== CONT  TestAccResilienceHubApp_tags_DefaultTags_nonOverlapping
--- PASS: TestAccResilienceHubApp_tags_DefaultTags_nonOverlapping (49.49s)
=== CONT  TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_emptyResourceTag
--- PASS: TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_emptyResourceTag (17.23s)
=== CONT  TestAccResilienceHubApp_tags_DefaultTags_providerOnly
--- PASS: TestAccResilienceHubApp_tags_DefaultTags_providerOnly (62.66s)
=== CONT  TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_updateToResourceOnly
--- PASS: TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_updateToResourceOnly (26.64s)
=== CONT  TestAccResilienceHubApp_tags_EmptyTag_OnUpdate_Replace
--- PASS: TestAccResilienceHubApp_tags_EmptyTag_OnUpdate_Replace (42.48s)
=== CONT  TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_updateToProviderOnly
--- PASS: TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_updateToProviderOnly (27.74s)
=== CONT  TestAccResilienceHubApp_tags_EmptyTag_OnUpdate_Add
--- PASS: TestAccResilienceHubApp_tags_EmptyTag_OnUpdate_Add (44.42s)
=== CONT  TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_overlapping
--- PASS: TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_overlapping (44.11s)
=== CONT  TestAccResilienceHubApp_tags_EmptyTag_OnCreate
--- PASS: TestAccResilienceHubApp_tags_EmptyTag_OnCreate (33.80s)
=== CONT  TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_nonOverlapping
--- PASS: TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_nonOverlapping (44.48s)
=== CONT  TestAccResilienceHubApp_tags_AddOnUpdate
--- PASS: TestAccResilienceHubApp_tags_AddOnUpdate (29.99s)
=== CONT  TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_providerOnly
--- PASS: TestAccResilienceHubResiliencyPolicy_tags_DefaultTags_providerOnly (58.65s)
=== CONT  TestAccResilienceHubApp_tags_EmptyMap
--- PASS: TestAccResilienceHubApp_tags_EmptyMap (19.56s)
=== CONT  TestAccResilienceHubResiliencyPolicy_tags_EmptyTag_OnUpdate_Replace
--- PASS: TestAccResilienceHubResiliencyPolicy_tags_EmptyTag_OnUpdate_Replace (27.93s)
=== CONT  TestAccResilienceHubApp_tags_null
--- PASS: TestAccResilienceHubApp_tags_null (18.64s)
=== CONT  TestAccResilienceHubResiliencyPolicy_tags_EmptyTag_OnUpdate_Add
--- PASS: TestAccResilienceHubResiliencyPolicy_tags_EmptyTag_OnUpdate_Add (40.99s)
=== CONT  TestAccResilienceHubResiliencyPolicy_tags_EmptyTag_OnCreate
--- PASS: TestAccResilienceHubResiliencyPolicy_tags_EmptyTag_OnCreate (30.39s)
=== CONT  TestAccResilienceHubResiliencyPolicy_upgradeV6_0_0
--- PASS: TestAccResilienceHubResiliencyPolicy_upgradeV6_0_0 (35.95s)
=== CONT  TestAccResilienceHubResiliencyPolicy_tags_AddOnUpdate
--- PASS: TestAccResilienceHubResiliencyPolicy_tags_AddOnUpdate (32.41s)
=== CONT  TestAccResilienceHubApp_complete
--- PASS: TestAccResilienceHubApp_complete (14.61s)
=== CONT  TestAccResilienceHubResiliencyPolicy_tags
--- PASS: TestAccResilienceHubResiliencyPolicy_tags (58.88s)
=== CONT  TestAccResilienceHubResiliencyPolicy_tags_EmptyMap
--- PASS: TestAccResilienceHubResiliencyPolicy_tags_EmptyMap (16.96s)
=== CONT  TestAccResilienceHubApp_update
--- PASS: TestAccResilienceHubApp_update (36.28s)
=== CONT  TestAccResilienceHubResiliencyPolicy_tags_null
--- PASS: TestAccResilienceHubResiliencyPolicy_tags_null (16.89s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/resiliencehub      1744.381s
```

briancain committed:
Feat: new resource aws_resiliencehub_app

Add the aws_resiliencehub_app resource for AWS Resilience Hub.

This resource manages the Resilience Hub application container itself —
name, description, assessment schedule, resiliency policy association,
and tags — handled entirely with AutoFlex.

Application version content (the app template, imported resource
sources, and publishing) is intentionally out of scope for this resource
and will be managed by a separate resource. AWS requires imported
resource sources before an application version can be published
(PublishAppVersion fails with "Application must have sources"), so the
template/import/publish lifecycle forms a distinct unit from the
application container.

Closes #35803.
Closes #25123.

jar-b commented on internal/service/resiliencehub/app.go:
```diff
@@ -0,0 +1,1542 @@
+			AppTemplateBody: aws.String(templateBody),
+		}
+
+		_, err = conn.PutDraftAppVersionTemplate(ctx, templateInput)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				create.ProblemStandardMessage(names.ResilienceHub, create.ErrActionCreating, ResNameApp, name, err),
+				err.Error(),
+			)
+			return
+		}
```
Should this be a separate resource, distinct from the parent app?

jar-b commented on internal/service/resiliencehub/app.go:
```diff
@@ -0,0 +1,1542 @@
+				TerraformSources: terraformSources,
+			}
+
+			_, err := conn.ImportResourcesToDraftAppVersion(ctx, importInput)
+			if err != nil {
+				resp.Diagnostics.AddError(
+					create.ProblemStandardMessage(names.ResilienceHub, create.ErrActionCreating, ResNameApp, name, err),
+					err.Error(),
+				)
+				return
+			}
```
The resource import workflow could also potentially be it's own resource.

jar-b commented on internal/service/resiliencehub/app.go:
```diff
@@ -0,0 +1,1542 @@
+				Optional: true,
+				Validators: []validator.String{
+					stringvalidator.OneOf("Daily", "Disabled"),
+				},
```
Any attributes with a `OneOf` validator can instead be represented with a `StringEnum` custom type.

jar-b commented on internal/service/resiliencehub/app.go:
```diff
@@ -0,0 +1,1542 @@
+	conn := r.Meta().ResilienceHubClient(ctx)
+
+	name := plan.Name.ValueString()
+	input := &resiliencehub.CreateAppInput{
+		Name: aws.String(name),
+		Tags: getTagsIn(ctx),
+	}
+
+	if !plan.Description.IsNull() {
+		input.Description = plan.Description.ValueStringPointer()
+	}
+
+	if !plan.AppAssessmentSchedule.IsNull() {
+		input.AssessmentSchedule = awstypes.AppAssessmentScheduleType(plan.AppAssessmentSchedule.ValueString())
+	}
+
+	if !plan.ResiliencyPolicyArn.IsNull() {
+		input.PolicyArn = plan.ResiliencyPolicyArn.ValueStringPointer()
+	}
```
AutoFlex should be used instead of setting each input explicitly.

jar-b commented on internal/service/resiliencehub/app.go:
```diff
@@ -0,0 +1,1542 @@
+		return
+	}
+
+	state.Arn = flex.StringToFramework(ctx, output.AppArn)
+	state.Description = flex.StringToFramework(ctx, output.Description)
+	state.Name = flex.StringToFramework(ctx, output.Name)
+	state.ResiliencyPolicyArn = flex.StringToFramework(ctx, output.PolicyArn)
```
Use AutoFlex to flatten the response.

jar-b reviewed (commented):
Left a few specific comments, but broadly the current implementation seems like it should be split into multiple resources. It might be worth completing a service design that describes potential resources, which APIs they map to, and the expected interaction between them.

briancain commented:
# AWS ResilienceHub App Resource - Terraform Schema Proposal

## Overview

This proposal introduces the `aws_resiliencehub_app` resource for Terraform, providing a structured interface that [**matches the CloudFormation experience**](https://docs.aws.amazon.com/AWSCloudFormation/latest/TemplateReference/aws-resource-resiliencehub-app.html) by combining both app template definition and resource mappings into a single, cohesive resource.

## Key Design Decision: Matching CloudFormation's Combined Approach

CloudFormation's `AWS::ResilienceHub::App` resource includes both:
- **AppTemplateBody**: A JSON string defining the application structure
- **ResourceMappings**: An array defining how logical resources map to physical resources

This Terraform implementation **replicates this exact functionality** but provides it in a more structured, type-safe way:
- **app_template block**: Structured equivalent of CloudFormation's `AppTemplateBody` JSON string
- **resource_mapping block**: Structured equivalent of CloudFormation's `ResourceMappings` array

This approach maintains **full feature parity with CloudFormation** while providing better Terraform ergonomics. Rather than defining a single string of loose json, this proposal allows users to craft the template as HCL.

## Schema Structure

### Basic Resource Configuration

```hcl
resource "aws_resiliencehub_app" "example" {
  name                    = "my-resilience-app"
  description             = "Example resilience hub application"
  app_assessment_schedule = "Daily"
  resiliency_policy_arn   = aws_resiliencehub_resiliency_policy.example.arn
  
  tags = {
    Environment = "production"
    Team        = "platform"
  }
}
```

### App Template Block (Replaces CloudFormation's AppTemplateBody JSON)

The `app_template` block provides a structured way to define what CloudFormation requires as a JSON string:

```hcl
app_template {
  version = "2.0"
  
  # Define application resources
  resource {
    name = "lambda-function"
    type = "AWS::Lambda::Function"
    
    logical_resource_id {
      identifier         = "MyLambdaFunction"
      logical_stack_name = "my-stack"
    }
  }
  
  # Define application components
  app_component {
    name           = "compute-tier"
    type           = "AWS::ResilienceHub::ComputeAppComponent"
    resource_names = ["lambda-function"]
  }
}
```

### Resource Mapping Block (Replaces CloudFormation's ResourceMappings Array)

The `resource_mapping` block provides the same functionality as CloudFormation's ResourceMappings but in a structured format:

```hcl
resource_mapping {
  mapping_type  = "Terraform"
  resource_name = "lambda-function"
  terraform_source_name = "my-terraform-source"
  
  physical_resource_id {
    type       = "Native"
    identifier = "s3://my-terraform-bucket/terraform.tfstate"
  }
}
```

## CloudFormation vs Terraform Comparison

### CloudFormation (JSON string approach):
```yaml
Type: AWS::ResilienceHub::App
Properties:
  Name: my-app
  AppTemplateBody: '{"resources":[{"logicalResourceId":{"identifier":"MyLambda"},"type":"AWS::Lambda::Function","name":"lambda-function"}],"appComponents":[{"name":"compute-tier","type":"AWS::ResilienceHub::ComputeAppComponent","resourceNames":["lambda-function"]}],"version":"2.0"}'
  ResourceMappings:
    - ResourceName: lambda-function
      MappingType: Terraform
      PhysicalResourceId:
        Type: Native
        Identifier: s3://my-bucket/terraform.tfstate
```

### Terraform (structured approach):
```hcl
resource "aws_resiliencehub_app" "my_app" {
  name = "my-app"
  
  app_template {
    version = "2.0"
    
    resource {
      name = "lambda-function"
      type = "AWS::Lambda::Function"
      
      logical_resource_id {
        identifier = "MyLambda"
      }
    }
    
    app_component {
      name           = "compute-tier"
      type           = "AWS::ResilienceHub::ComputeAppComponent"
      resource_names = ["lambda-function"]
    }
  }
  
  resource_mapping {
    mapping_type  = "Terraform"
    resource_name = "lambda-function"
    
    physical_resource_id {
      type       = "Native"
      identifier = "s3://my-bucket/terraform.tfstate"
    }
  }
}
```

**Result**: Same functionality, better developer experience.

## Complete Example: Terraform Source Integration

```hcl
resource "aws_resiliencehub_app" "terraform_app" {
  name                    = "terraform-managed-app"
  description             = "App with S3 Terraform state source"
  app_assessment_schedule = "Daily"
  
  app_template {
    version = "2.0"
    
    resource {
      name = "lambda-function"
      type = "AWS::Lambda::Function"
      
      logical_resource_id {
        identifier            = "MyLambda"
        terraform_source_name = "my-terraform-source"
      }
    }
    
    app_component {
      name           = "compute-tier"
      type           = "AWS::ResilienceHub::ComputeAppComponent"
      resource_names = ["lambda-function"]
    }
  }
  
  resource_mapping {
    mapping_type          = "Terraform"
    resource_name         = "lambda-function"
    terraform_source_name = "my-terraform-source"
    
    physical_resource_id {
      type       = "Native"
      identifier = "s3://my-terraform-bucket/terraform.tfstate"
    }
  }
}
```

## Schema Attributes

### Top-level Attributes

| Attribute | Type | Required | Description |
|-----------|------|----------|-------------|
| `name` | `string` | Yes | Application name (2-60 chars, alphanumeric, underscore, hyphen) |
| `description` | `string` | No | Application description (max 500 chars) |
| `app_assessment_schedule` | `string` | No | Assessment schedule: "Daily" or "Disabled" |
| `resiliency_policy_arn` | `string` | No | ARN of the resiliency policy |
| `tags` | `map(string)` | No | Resource tags |

### App Template Block

| Attribute | Type | Required | Description |
|-----------|------|----------|-------------|
| `version` | `string` | Yes | Template version (e.g., "2.0") |
| `additional_info` | `map(string)` | No | Additional template metadata |

#### Resource Sub-block

| Attribute | Type | Required | Description |
|-----------|------|----------|-------------|
| `name` | `string` | Yes | Logical resource name |
| `type` | `string` | Yes | AWS resource type |
| `additional_info` | `map(string)` | No | Additional resource metadata |

#### Logical Resource ID Sub-block

| Attribute | Type | Required | Description |
|-----------|------|----------|-------------|
| `identifier` | `string` | Yes | Resource identifier |
| `logical_stack_name` | `string` | No | CloudFormation stack name |
| `resource_group_name` | `string` | No | Resource group name |
| `terraform_source_name` | `string` | No | Terraform source identifier |
| `eks_source_name` | `string` | No | EKS source identifier |

#### App Component Sub-block

| Attribute | Type | Required | Description |
|-----------|------|----------|-------------|
| `name` | `string` | Yes | Component name |
| `type` | `string` | Yes | Component type (e.g., "AWS::ResilienceHub::ComputeAppComponent") |
| `resource_names` | `list(string)` | No | List of resource names in this component |
| `additional_info` | `map(string)` | No | Additional component metadata |

### Resource Mapping Block

| Attribute | Type | Required | Description |
|-----------|------|----------|-------------|
| `mapping_type` | `string` | Yes | Mapping type: "Terraform", "Resource", "AppRegistry", "CfnStack", "EKS" |
| `resource_name` | `string` | Yes | Name of the logical resource |
| `terraform_source_name` | `string` | No | Terraform source name (for Terraform mappings) |

#### Physical Resource ID Sub-block

| Attribute | Type | Required | Description |
|-----------|------|----------|-------------|
| `type` | `string` | Yes | Identifier type: "Arn", "Native" |
| `identifier` | `string` | Yes | Physical resource identifier or S3 URL |
| `aws_account_id` | `string` | No | AWS account ID |
| `aws_region` | `string` | No | AWS region |

## Implementation Benefits

1. **CloudFormation Feature Parity**: Exact same functionality as CloudFormation's `AWS::ResilienceHub::App`
2. **Better Developer Experience**: Structured blocks instead of JSON strings
3. **Type Safety**: Built-in validation and IDE support
4. **Terraform Integration**: Native support for Terraform state file sources
5. **Maintainability**: Easier to read, write, and maintain than JSON strings

## Test Coverage Examples

### Basic App Creation
```hcl
resource "aws_resiliencehub_app" "test" {
  name                    = "tf-test-app"
  app_assessment_schedule = "Disabled"

  app_template {
    version = "2.0"

    app_component {
      name           = "appcommon"
      type           = "AWS::ResilienceHub::AppCommonAppComponent"
      resource_names = []
    }
  }
}
```

### Terraform Source Integration
```hcl
resource "aws_resiliencehub_app" "test" {
  name                    = "terraform-source-app"
  description             = "Test app with S3 Terraform source"
  app_assessment_schedule = "Disabled"

  app_template {
    version = "2.0"

    resource {
      name = "lambda-function"
      type = "AWS::Lambda::Function"

      logical_resource_id {
        identifier            = "MyLambda"
        terraform_source_name = "my-terraform-source"
      }
    }

    app_component {
      name           = "compute-tier"
      type           = "AWS::ResilienceHub::ComputeAppComponent"
      resource_names = ["lambda-function"]
    }
  }

  resource_mapping {
    mapping_type          = "Terraform"
    resource_name         = "lambda-function"
    terraform_source_name = "my-terraform-source"

    physical_resource_id {
      type       = "Native"
      identifier = "s3://my-terraform-bucket/terraform.tfstate"
    }
  }
}
```

### Complete Multi-Component App
```hcl
resource "aws_resiliencehub_app" "test" {
  name                    = "complete-app"
  description             = "Complete test app"
  app_assessment_schedule = "Daily"

  app_template {
    version = "2.0"

    resource {
      name = "lambda-function"
      type = "AWS::Lambda::Function"

      logical_resource_id {
        identifier         = "MyLambda"
        logical_stack_name = "my-stack"
      }
    }
    
    resource {
      name = "database"
      type = "AWS::RDS::DBInstance"

      logical_resource_id {
        identifier         = "MyDatabase"
        logical_stack_name = "my-stack"
      }
    }

    app_component {
      name           = "compute-tier"
      type           = "AWS::ResilienceHub::ComputeAppComponent"
      resource_names = ["lambda-function"]
    }

    app_component {
      name           = "database-tier"
      type           = "AWS::ResilienceHub::DatabaseAppComponent"
      resource_names = ["database"]
    }
  }

  tags = {
    Environment = "test"
    Purpose     = "testing"
  }
}
```

gdavison commented on internal/service/resiliencehub/app.go:
```diff
@@ -0,0 +1,1502 @@
+		plan.DriftStatus = types.StringValue(string(app.DriftStatus))
+	} else {
+		plan.DriftStatus = types.StringValue("NotChecked")
+	}
```
The value of `drift_status` should be set to whatever the AWS API returns. It will never be unknown

```suggestion
	plan.DriftStatus = types.StringValue(string(app.DriftStatus))
```

gdavison commented on internal/service/resiliencehub/app.go:
```diff
@@ -0,0 +1,1502 @@
+		return nil, &retry.NotFoundError{
+			LastError:   err,
+			LastRequest: input,
+		}
```
Prefer `retry.NotFoundError` from `github.com/hashicorp/terraform-provider-aws/internal/retry`

```suggestion
		return nil, &retry.NotFoundError{
			LastError: err,
		}
```

gdavison commented on internal/service/resiliencehub/testdata/tmpl/app_tags.gtpl:
```diff
@@ -0,0 +1,30 @@
+resource "aws_resiliencehub_app" "test" {
```
Typically, the configuration for the tagging test should be as simple as possible, often only using required attributes.

gdavison commented on internal/service/resiliencehub/app.go:
```diff
@@ -0,0 +1,1502 @@
+		resp.Diagnostics.AddError("Failed to list tags", err.Error())
+		return
+	}
+	setTagsOut(ctx, tags.Map())
```
Since there is a `Tags` field on the `awstypes.App` returned by `FindAppByARN`, a separate `listTags` call is not needed, instead use the following:

```suggestion
	setTagsOut(ctx, output.Map())
```

gdavison commented on website/docs/r/resiliencehub_app.html.markdown:
````diff
@@ -0,0 +1,239 @@
+
+### Complete Usage with Resources and Terraform Source
+
+```terraform
````
The resource that we're documenting should be the first resource in the example.

gdavison reviewed (changes requested):
Hi @briancain, in addition to @jar-b's earlier review, I've taken a look at the more structural elements of your PR. I have a number of suggested changes.

Additionally, the [Contributor Guide for the provider](https://hashicorp.github.io/terraform-provider-aws/) has comprehensive information on development on this provider.

david-forster10 commented:
@briancain As a interested 3rd party in this resource type to be added to the provider, will you be providing functionality for defining excluded resources within the resources under the app template definition?

Value is `excludedResources` referenced in the appTemplateBody API here:

https://docs.aws.amazon.com//resilience-hub/latest/APIReference/API_PutDraftAppVersionTemplate.html#API_PutDraftAppVersionTemplate_RequestSyntax

gdavison commented on internal/service/resiliencehub/app.go:
```diff
@@ -0,0 +1,1487 @@
+// @Tags(identifierAttribute="arn")
+// @Testing(existsType="github.com/aws/aws-sdk-go-v2/service/resiliencehub;resiliencehub.DescribeAppOutput")
+// @Testing(importStateIdAttribute="arn")
+// @Testing(preIdentityVersion="v5.100.0")
```
Since this is a new resource type, the annotation should be

```suggestion
// @Testing(hasNoPreExistingResource=true)
```

The annotation `preIdentityVersion="v5.100.0"` means that Resource Identity was added to the resource type in v6.0.0 of the provider.

gdavison commented on internal/service/resiliencehub/testdata/App/basic_v5.100.0/main_gen.tf:
This resource type did not exist in v5.100.0. This file should be removed

gdavison commented on internal/service/resiliencehub/testdata/App/basic_v6.0.0/main_gen.tf:
This resource type did not exist in v6.0.0. This file should be removed

gdavison commented on internal/service/resiliencehub/testdata/tmpl/app_tags.gtpl:
```diff
@@ -0,0 +1,30 @@
+resource "aws_resiliencehub_app" "test" {
```
The test `TestAccResilienceHubApp_Identity_RegionOverride` will continue to fail if this is not added

gdavison commented on internal/service/resiliencehub/testdata/tmpl/app_tags.gtpl:
This file will need to be renamed to `app_basic.gtpl` before the branch can be merged. Either rebase the PR or merge the `main` branch, rename the file and re-run the generators

gdavison commented on internal/service/resiliencehub/app.go:
```diff
@@ -0,0 +1,1487 @@
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(regexache.MustCompile(`^[A-Za-z0-9][A-Za-z0-9_\-]{1,59}$`), "must be 2-60 characters, start with alphanumeric, contain only alphanumeric, underscore, and hyphen"),
```
The string length and regex should be validated separately, e.g.

```suggestion
					stringvalidator.LengthBetween(2, 60),
					stringvalidator.RegexMatches(regexache.MustCompile(`^[A-Za-z0-9][A-Za-z0-9_\-]$`), "must start with alphanumeric, contain only alphanumeric, underscore, and hyphen"),
```

gdavison commented on internal/service/resiliencehub/app.go:
```diff
@@ -0,0 +1,1487 @@
+			},
+			"resiliency_policy_arn": schema.StringAttribute{
+				Optional: true,
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(regexache.MustCompile(`^arn:(aws|aws-cn|aws-iso|aws-iso-[a-z]{1}|aws-us-gov):[A-Za-z0-9][A-Za-z0-9_/.-]{0,62}:([a-z]{2}-((iso[a-z]{0,1}-)|(gov-))?[a-z]+-[0-9]):[0-9]{12}:[A-Za-z0-9][A-Za-z0-9:_/+=,@.-]{0,255}$`), "must be a valid ARN"),
+				},
+			},
```
This should use the `fwtypes.ARNType` custom type. The model will also need to be updated.


```suggestion
			"resiliency_policy_arn": schema.StringAttribute{
				CustomType: fwtypes.ARNType,
				Optional:   true,
			},
```

gdavison commented on internal/service/resiliencehub/app.go:
```diff
@@ -0,0 +1,1487 @@
+
+	r.SetDefaultCreateTimeout(30 * time.Minute)
+	r.SetDefaultUpdateTimeout(30 * time.Minute)
+	r.SetDefaultDeleteTimeout(30 * time.Minute)
```
These seem high compared to how long the acceptance tests take to run

gdavison commented on internal/service/resiliencehub/app_test.go:
```diff
@@ -0,0 +1,464 @@
+  })
+}
+
+resource "aws_resiliencehub_app" "test" {
```
For ease of reading, move the resource type under test to the top of the configuration

gdavison commented on internal/service/resiliencehub/app_test.go:
```diff
@@ -0,0 +1,464 @@
+	})
+}
+
+func TestAccResilienceHubApp_complete(t *testing.T) {
```
Instead of a `_complete` test, add attribute-specific tests

gdavison commented on internal/service/resiliencehub/app.go:
```diff
@@ -0,0 +1,1502 @@
+type appResourceModel struct {
+	AppArn             types.String                                           `tfsdk:"arn"`
+	AssessmentSchedule fwtypes.StringEnum[awstypes.AppAssessmentScheduleType] `tfsdk:"app_assessment_schedule"`
+	AppTemplate        types.List                                             `tfsdk:"app_template" autoflex:"-"`
```
I'd suggest following a pattern like we use for `aws_iam_policy` and `aws_iam_policy_document`. This allows users the flexibility of using an existing JSON document and allows them to either use the Terraform function [`jsonencode`](https://developer.hashicorp.com/terraform/language/functions/jsonencode) or a data source that handles defaults.

Following the underlying API is one of the Terraform provider design principles: https://developer.hashicorp.com/terraform/plugin/best-practices/hashicorp-provider-design-principles#resource-and-attribute-schema-should-closely-match-the-underlying-api

gdavison commented on internal/service/resiliencehub/app_test.go:
```diff
@@ -0,0 +1,464 @@
+		PreCheck: func() {
+			acctest.PreCheck(ctx, t)
+		},
+		ErrorCheck:               acctest.ErrorCheck(t, names.ResilienceHubServiceID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckAppDestroy(ctx),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAppConfig_basic(rName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckAppExists(ctx, resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, names.AttrName, rName),
+					acctest.MatchResourceAttrRegionalARN(ctx, resourceName, names.AttrARN, "resiliencehub", regexache.MustCompile(`app/.+`)),
+				),
```
The `_basic` test should check the values of all attributes

gdavison commented on website/docs/r/resiliencehub_app.html.markdown:
```diff
@@ -0,0 +1,239 @@
+
+* `arn` - ARN of the application.
+* `drift_status` - Drift status of the application.
+* `id` - ARN of the application.
```
There is no `id` attribute, so this should be removed

gdavison commented on internal/service/resiliencehub/app_test.go:
```diff
@@ -0,0 +1,464 @@
+	rName := acctest.RandomWithPrefix(t, acctest.ResourcePrefix)
+	resourceName := "aws_resiliencehub_app.test"
+
+	resource.ParallelTest(t, resource.TestCase{
```
`resource.ParallelTest` should be replaced with `acctest.ParallelTest`. This will fail code checks when merged with the `main` branch unless it is updated. Either rebase the PR or merge the main branch, and run `make semgrep-fix PKG=resiliencehub`

gdavison commented on internal/service/resiliencehub/app.go:
```diff
@@ -0,0 +1,1487 @@
+			"assessment_schedule": schema.StringAttribute{
+				CustomType: fwtypes.StringEnumType[awstypes.AppAssessmentScheduleType](),
+				Optional:   true,
+				Computed:   true,
```
Since this is not expected to change unless modified by the user, it should use the current state if no value is set

```suggestion
				Computed:   true,
				PlanModifiers: []planmodifier.String{
					stringplanmodifier.UseStateForUnknown(),
				},
```

gdavison commented on internal/service/resiliencehub/app_test.go:
```diff
@@ -0,0 +1,464 @@
+
+func TestAccResilienceHubApp_update(t *testing.T) {
+	ctx := acctest.Context(t)
+	var v resiliencehub.DescribeAppOutput
+	rName := acctest.RandomWithPrefix(t, acctest.ResourcePrefix)
+	resourceName := "aws_resiliencehub_app.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acctest.PreCheck(ctx, t)
+		},
+		ErrorCheck:               acctest.ErrorCheck(t, names.ResilienceHubServiceID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckAppDestroy(ctx),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAppConfig_basic(rName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckAppExists(ctx, resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, names.AttrName, rName),
+				),
+			},
+			{
+				Config: testAccAppConfig_updateTemplate(rName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckAppExists(ctx, resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, names.AttrName, rName),
+					resource.TestCheckResourceAttr(resourceName, names.AttrDescription, "Updated description"),
+					resource.TestCheckResourceAttr(resourceName, "app_template.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "app_template.0.version", "2.0"),
+				),
+			},
+			{
+				Config: testAccAppConfig_updateResourceMapping(rName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckAppExists(ctx, resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "resource_mapping.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "resource_mapping.0.mapping_type", "Terraform"),
+				),
+			},
```
Updating the template, the description, and the resource mapping should be three separate tests. The before and after steps should clearly demonstrate that the value has changed by showing the values before and after the update.

gdavison commented on internal/service/resiliencehub/app.go:
```diff
@@ -0,0 +1,1487 @@
+	conn := r.Meta().ResilienceHubClient(ctx)
+
+	name := plan.Name.ValueString()
+	input := &resiliencehub.CreateAppInput{
```
The idempotency token `ClientToken` should be set for the Create, Update, and Delete API calls

gdavison reviewed (changes requested):
Thanks for your patience, @briancain 

I have a number of suggested changes attached.

The resource design you included at https://github.com/hashicorp/terraform-provider-aws/pull/44076#issuecomment-3286800704 describes the **what** of the design, but what @jar-b and I would like to see is the **why**. Some things to consider in the design:

* Design parity with CloudFormation is not a design goal for the provider
* An App will have one or more App Versions, which can be in draft or published state
* Resources and Application Components can be added to draft application versions
* To get most use out of an App, it should be published. It doesn't appear that the current resource design publishes the App Version
* Event subscriptions and the permission model aren't used. Is that intended for future addition?

briancain committed:
Feat: new resource aws_resiliencehub_app_version

Add the aws_resiliencehub_app_version resource for AWS Resilience Hub.

This resource defines and publishes a Resilience Hub application version.
It sets the draft application template (app_template_body, a JSON string
mirroring the PutDraftAppVersionTemplate API, in the style of aws_iam_policy),
imports resource sources (CloudFormation stacks via source_arns or Terraform
state files via terraform_source), waits for the import to complete, and
publishes a new application version.

briancain commented:
Thanks for the reviews, @gdavison and @jar-b! I've rebased onto the latest `main`, fixed the merge conflicts, and reworked the feature to use AutoFlex throughout, so the implementation has diverged a fair bit from my year-old schema proposal above. Let me know what else you'd like changed.

~edit: Also note that the current Semgrep failure is not something I've touched, looks to be a failure from `main`?~

briancain commented:
Hi @gdavison @jar-b - what's the status of reviewing this PR?
